pages/rfc/sso-rfc.rst

```diff
@@ -0,0 +1,230 @@
+to the client so it can authorize itself to a resource server. The contents of
+the token is opaque to the client. This means that the client does not need to
+know anything about the content or structure of the token itself.
+
+The resource server can exchange the Access Token at its own Authorization Server
+for a JSON document containing all the needed attributes.
+
+Authorization server
+====================
+
+The Authorization server checks the user's identity and credentials and creates the
+access token which should be used during the jump. The authorization server is
+trusted by the resource server. The resource server can exchange the access token
+for a JSON document with the user's identity, subject, custodian, validity and scopes.
+This mechanism called token introspection and is is described in `rfc7662 <https://tools.ietf.org/html/rfc7662>`_
+
+Mechanics
+*********
+
+We have two applications: The client application, and the resource server.
+Both run their own Nuts Node (could be the same node). The client application collects
+information about the user and patient, acquires an access token by posting a
+JWT Bearer Token to an authorization server (this can be the Nuts node or a
+custom implementation).
+With this access token the user jumps to the resource server. The resource server
+posts the access token to its authorization server to retrieve the contents of
+the access token.
+
+SSO Steps
+=========
+
+#. The User loads a page in the client application in the context of a certain subject (patient).
+#. The client application checks rights and settings to see if this user is allows to jump (local policy)
+#. The client application checks if the subject has any external custodians (care providers). This subject <-> custodian relationship should be registered locally.
+#. For each custodian the client application checks if it has a jump endpoint
+   The endpoint is described in the :ref:`nuts-registry-api`
+
+   .. code-block:: console
+
+     $ curl --location --request GET 'localhost:1323/api/endpoints?orgIds=urn:oid:2.16.840.1.113883.2.4.6.1:00000001&type=urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint'
+
+       [{
+          "URL": "http://sso.nootenboom.local",
+          "endpointType": "urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint",
+          "identifier": "a5c4afba-1393-48dc-b506-e0b6cc969094",
+          "status": "active",
+          "version": "0.1.0"
+        }]
+
+#. The client application renders a SSO button
+#. The User clicks the button
+#. The client application collects the users identity using a login contract. If not already present, it lets the user sign one using IRMA
+   To create an IRMA session, make a call to the Nuts Auth server as described in the :ref:`nuts-consent-auth-api`
+
+   .. code-block:: console
+
+     $ curl --location --request POST 'localhost:1323/auth/contract/session' \
```

Comment:
Comment about future addition for adding scope?

Reply:
👍 I will add a `scope` field to the `CreateJwtBearerToken` request.